addons/event-audit/common/src/main/java/org/commonjava/indy/event/audit/EventAuditListener.java

```diff
@@ -0,0 +1,227 @@
+        eventPublisher.publishFileGroupingEvent( fileGroupingEvent );
+    }
+
+    private void transformFileEvent( org.commonjava.maven.galley.event.FileEvent event, FileEvent fileEvent )
+    {
+        EventMetadata metadata = event.getEventMetadata();
+        final TrackingKey trackingKey = (TrackingKey) metadata.get( "tracking-id" );
+        if ( trackingKey == null )
+        {
+            logger.trace( "No tracking key. Skip audit." );
+            return;
+        }
+
+        Transfer transfer = event.getTransfer();
+        if ( transfer == null )
+        {
+            logger.trace( "No transfer. Skip audit." );
+            return;
+        }
+
+        fileEvent.setTargetPath( transfer.getPath() );
+        fileEvent.setNodeId( "" );
+        fileEvent.setSessionId( trackingKey.getId() );
+        fileEvent.setTimestamp( new Date() );
+
+        final KeyedLocation keyedLocation = (KeyedLocation) transfer.getLocation();
+        final String path = transfer.getPath();
+        try
+        {
+            StoreKey affectedStore = keyedLocation.getKey();
+            final Transfer txfr = downloadManager.getStorageReference( keyedLocation.getKey(), transfer.getPath() );
+            if ( txfr != null )
+            {
+
+                if ( StoreType.remote == affectedStore.getType() )
+                {
+                    final RemoteRepository repo = (RemoteRepository) storeManager.getArtifactStore( affectedStore );
+                    if ( repo != null )
+                    {
+                        fileEvent.setTargetLocation( repo.getUrl() );
+                    }
+                }
+
+                TransferMetadata artifactData = contentDigester.digest( affectedStore, path, event.getEventMetadata() );
+                Map<String, String> extra = new HashMap<>();
+                extra.put( ContentDigest.MD5.name(), artifactData.getDigests().get( ContentDigest.MD5 ) );
+                extra.put( ContentDigest.SHA_1.name(), artifactData.getDigests().get( ContentDigest.SHA_1 ) );
+                extra.put( ContentDigest.SHA_256.name(), artifactData.getDigests().get( ContentDigest.SHA_256 ) );
+                extra.put( "Size", String.valueOf( artifactData.getSize() ) );
+
+                fileEvent.setChecksum( artifactData.getDigests().get( ContentDigest.MD5 ) );
+
+                fileEvent.setExtra( extra );
+            }
+
+        }
+        catch ( final IndyWorkflowException | IndyDataException e )
+        {
+            logger.error( String.format( "Failed to transform file event. Reason: %s", e.getMessage() ), e );
+        }
+    }
+
+    private void transformFileGroupingEvent( PromoteCompleteEvent event, FileGroupingEvent fileGroupingEvent )
+    {
+
+        PathsPromoteCompleteEvent pathsPromoteCompleteEvent = null;
+        if ( event instanceof PathsPromoteCompleteEvent )
+        {
+            pathsPromoteCompleteEvent = (PathsPromoteCompleteEvent) event;
+        }
```

Comment:
No else? I know currently only one type of PromoteCompleteEvent is Paths, but I think we still need to prevent NPE here if some new types added in future.

Reply:
Good point.